Reconstruction/doc/RecCalorimeter.md

```diff
@@ -3,51 +3,122 @@ RecCalorimeter package
 * readoutName (string, name of the detector readout)
-* activeVolumeName (string, name of active volumes - material name)
-* numVolumesRemove (unsigned, number of volumes with active material which are not readout - e.g. ECAL: LAr bath in cryostat)
+* activeVolumeName (string, name of active volumes)
 * activeFieldName (string, name of active layers for sampling calorimeter)
-* fieldNames (vector of string, name of the fields describing the segmented volume)
-* fieldValues (vector of int, values of the fields describing the segmented volume);
+* fieldNames (list of strings, name of the fields describing the segmented volume)
+* fieldValues (list of ints, values of the fields describing the segmented volume)
+
+## 1. Deposits merging
+
+The output of the simulation contains a large collection of energy deposits. First step towards the digitisation is merging those hits into the hit collection with one hit per cell of the sensitive volume. The energy of such hit is a sum of all the deposits within that sensitive volume cell. 
+
+> Note: Energy deposits may be merged by Sensitive Detector implementation (`AggregateCalorimeterSD`), however it is not encouraged due to the time performance issues.
+
+>TODO: Take time into account. At the moment the time of the last hit in the cell is stored.
+
+`MergeCaloHitsTool`: Merge Geant4 energy deposits with same cellID
+
+## 2. Energy calibration
+
+Energy calibration is done for sampling calorimeters to compensate for the energy deposited in the passive material.
+
+`CalibrateCaloHitsTool`: Calibrate Geant4 energy to electromagnetic (EM) scale. The calibration constant (inverse sampling fraction) is a configurable parameter of this tool.
+
+## 3. Noise addition
+
+Noise is added to all cells, not only those with signal. Hence a list of all cellIDs is needed and it is retrieved from the geometry. This is the only part which is geometry dependent and assumes tube geometry and phi-eta segmentation.
+
+ `NoiseCaloCellsTool`: Prepare random noise hits for each cell.
+ `MergeCaloHitsTool`: Merge signal with noise hits.
+ `NoiseCaloCellsTool`: Filter cell energy and accept only cells with energy above threshold.
+
 
-More details can be found in [Doxygen documentation] (http://fccsw.web.cern.ch/fccsw/).
+# Reconstruction
+
+Reconstruction creates clusters (`fcc::CaloCluster`) out of cells (`fcc::CaloHit`). Each cluster stores the information about its global position (x, y, z), energy and the relation to the cells it is composed of.
+
+## Sliding window algorithm
+
+Sliding window algorithm is an algorithm used for the reconstruction of electrons and photons [ref].
+
+`CreateCaloClustersSlidingWindow` implementation assumes the detector with eta-phi segmentation.
+
+> Note: If this segmentation was not used for the original simulation and the true deposits' positions are saved (`fcc::PositionedCaloHit`, it is possible to resegment using `RedoSegmentation` algorithm.
+
+Options:
```

Comment:
I'd prefer if we only document the options in the algorithm, that makes maintaining the documentation easier.